test/new-e2e/tests/agent-platform/persisting-integrations/persisting_integrations_test.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/platforms"
+
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TODO: This is a temporary workaround to test the upgrade of the persisting integrations
+// Since we need the previous agent version to have produced these files
+// We will mock these files for now since this is the first release that will generate them
+// Once we have a release that generates these files we can remove this workaround and stop mocking these files
+
+//go:embed fixtures/diff.txt
+var diffPythonInstalledPackages string
+
+//go:embed fixtures/postinst.txt
+var postinstPythonInstalledPackages string
+
+//go:embed fixtures/requirements.txt
+var requirementsAgentRelease string
+
+var (
+	osVersion       = flag.String("osversion", "", "os version to test")
+	platform        = flag.String("platform", "", "platform to test")
+	architecture    = flag.String("arch", "", "architecture to test (x86_64, arm64))")
+	flavorName      = flag.String("flavor", "datadog-agent", "package flavor to install")
+	srcAgentVersion = flag.String("src-agent-version", "7", "start agent version")
+)
+
+var filesToMock = map[string]string{
+	"/opt/datadog-agent/.diff_python_installed_packages.txt":     diffPythonInstalledPackages,
+	"/opt/datadog-agent/.postinst_python_installed_packages.txt": postinstPythonInstalledPackages,
+	"/opt/datadog-agent/requirements-agent-release.txt":          requirementsAgentRelease,
+}
+
+type persistingIntegrationsSuite struct {
+	e2e.BaseSuite[environments.Host]
+	srcVersion string
+	platform   string
+}
+
+func (is *persistingIntegrationsSuite) AfterTest(suiteName, testName string) {
+	is.BaseSuite.AfterTest(suiteName, testName)
+	platform := strings.ToLower(is.platform)
+
+	if platform == "ubuntu" || platform == "debian" {
+		is.Env().RemoteHost.Execute("sudo apt-get remove datadog-agent -y")
+		is.Env().RemoteHost.MustExecute("sudo apt-get remove --purge datadog-agent -y")
+	} else if platform == "redhat" || platform == "amazonlinux" || platform == "centos" {
+		is.Env().RemoteHost.MustExecute("sudo yum remove datadog-agent -y")
+		is.Env().RemoteHost.Execute("sudo userdel dd-agent")
+		is.Env().RemoteHost.Execute("sudo rm -rf /opt/datadog-agent/")
+		is.Env().RemoteHost.Execute("sudo rm -rf /etc/datadog-agent/")
+		is.Env().RemoteHost.Execute("sudo rm -rf /var/log/datadog/")
+	} else if platform == "suse" {
+		is.Env().RemoteHost.MustExecute("sudo zypper remove datadog-agent")
+		is.Env().RemoteHost.Execute("sudo userdel dd-agent")
+		is.Env().RemoteHost.Execute("sudo rm -rf /opt/datadog-agent/")
+		is.Env().RemoteHost.Execute("sudo rm -rf /etc/datadog-agent/")
+		is.Env().RemoteHost.Execute("sudo rm -rf /var/log/datadog/")
+	} else {
+		is.T().Fatal("Unsupported platform for cleanup")
+	}
+}
+
+func TestPersistingIntegrations(t *testing.T) {
+	platformJSON := map[string]map[string]map[string]string{}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I would move the skip up here:

```suggestion
	flake.Mark(t)
	platformJSON := map[string]map[string]map[string]string{}
```

Reply:
Done!